CMakeLists.txt

```diff
@@ -0,0 +1,227 @@
+# Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+# CMake build rules for SustainML Front-end
+###############################################################################
+cmake_minimum_required(VERSION 3.5)
+cmake_policy(VERSION 3.5...3.13)
+
+# Done this to set machine architecture and be able to call cmake_utils
+enable_language(CXX)
+
+###############################################################################
+# Find package cmake_utils
+###############################################################################
+# Package cmake_utils is required to get every cmake macro needed
+find_package(cmake_utils REQUIRED)
+
+###############################################################################
+# Project
+###############################################################################
+# Configure project by info set in project_settings.cmake
+# - Load project_settings variables
+# - Read version
+# - Set installation paths
+configure_project()
+
+# Call explictly project
+project(
+    ${MODULE_NAME}
+    VERSION
+        ${MODULE_VERSION}
+    DESCRIPTION
+        ${MODULE_DESCRIPTION}
+    LANGUAGES
+        CXX
+)
+
+###############################################################################
+# C++ Project
+###############################################################################
+# Configure CPP project for dependencies and required flags:
+# - Set CMake Build Type
+# - Set C++ version
+# - Set shared libraries by default
+# - Find external packages and thirdparties
+# - Activate Code coverage if flag CODE_COVERAGE
+# - Activate Address sanitizer if flag ASAN_BUILD
+# - Configure log depending on LOG_INFO flag and CMake type
+configure_project_cpp()
+
+###############################################################################
+# Load external projects
+###############################################################################
+set(QT_PATH "" CACHE STRING "Path to the Qt installation folder")
+list(APPEND CMAKE_PREFIX_PATH ${QT_PATH})
+
+find_package(Qt5 5.15 COMPONENTS Core Widgets Gui Qml Quick Charts QuickControls2 REQUIRED)
+find_package(fastcdr REQUIRED)
```

Comment:
I think you don't need all of this if you use `cmake_utils` to build the project

Reply:
This will be addressed in other PR